### PR TITLE
Zip Deploy - Archive Folder package

### PR DESCRIPTION
**Pull request by vincentdass**

### CODE REVIEW

**Comment by vincentdass** [comment]
#8110

File: Tasks/AzureRmWebAppDeploymentV4/operations/KuduServiceUtility.ts [hunk 1]
```diff
@@ -158,6 +158,12 @@ export class KuduServiceUtility {
                 await webClient.sleepFor(5);
             }
 
+            if(tl.stats(packagePath).isDirectory()) {
```
Comment (kmkumaran) [inline Tasks/AzureRmWebAppDeploymentV4/operations/KuduServiceUtility.ts]: As discussed, please add test coverage for the folder path scenario.

File: Tasks/AzureRmWebAppDeploymentV4/operations/KuduServiceUtility.ts [hunk 2]
```diff
@@ -158,6 +158,12 @@ export class KuduServiceUtility {
...
+                let tempPackagePath = deployUtility.generateTemporaryFolderOrZipPath(tl.getVariable('AGENT.TEMPDIRECTORY'), false);
+                packagePath = await zipUtility.archiveFolder(packagePath, "", tempPackagePath);
+                tl.debug("Compressed folder " + packagePath + " into zip : " +  packagePath);
+            }
+
```
Comment (RoshanKumarMicrosoft) [inline Tasks/AzureRmWebAppDeploymentV4/operations/KuduServiceUtility.ts]: Can we move this logic inside provider ?

Comment (RoshanKumarMicrosoft) [reply Tasks/AzureRmWebAppDeploymentV4/operations/KuduServiceUtility.ts]: As Discussed you can move this logic in separate PR. As in existing Provider layer we are checking if it is folder don't zip it but in next layer Kudu service API call we are again checking if folder zip this folder.

Comment (vincentdass) [reply Tasks/AzureRmWebAppDeploymentV4/operations/KuduServiceUtility.ts]: thanks, moved the logic to deploymentProvider.

**Pull request closed without merging**